metricflow-semantics/tests_metricflow_semantics/query/test_query_parser.py

```diff
@@ -459,6 +466,52 @@ def test_cumulative_metric_agg_time_dimension_name_validation(
     assert_object_snapshot_equal(request=request, mf_test_configuration=mf_test_configuration, obj=result)
 
 
+def test_join_to_scd_no_time_dimension_validation(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    scd_query_parser: MetricFlowQueryParser,
+) -> None:
+    """Test that queries that join to SCD semantic models fail if no time dimensions are selected."""
+    with pytest.raises(InvalidQueryException) as exc_info:
+        scd_query_parser.parse_and_validate_query(
+            metric_names=["bookings"],
+            group_by_names=["listing__country"],
+        )
+
+    assert len(exc_info.value.args) == 1
+    assert isinstance(exc_info.value.args[0], str)
+    assert_str_snapshot_equal(
+        request=request,
+        mf_test_configuration=mf_test_configuration,
+        snapshot_id="error",
+        snapshot_str=exc_info.value.args[0],
+    )
+
+
+def test_join_through_scd_no_time_dimension_validation(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    scd_query_parser: MetricFlowQueryParser,
+) -> None:
+    """Test that queries that join through SCDs semantic models fail if no time dimensions are selected."""
+    with pytest.raises(InvalidQueryException) as exc_info:
```

Comment:
@courtneyholcomb I change these tests from using the `match` regex to use `assert_str_snapshot_equal`, and I created the 2 new `.txt` files with the full error.